### PR TITLE
Check setsockopt return

### PR DESCRIPTION
**Pull request by pablogs9**
Signed-off-by: Pablo Garrido <pablogs9@gmail.com>

### CODE REVIEW

**Comment by richiprosima** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/Micro-XRCE-DDS-Client%20Manual%20Linux/564/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/Micro-XRCE-DDS-Client%20Manual%20Windows/582/)

File: src/c/profile/transport/ip/udp/udp_transport_posix_nopoll.c [hunk 1]
```diff
@@ -102,7 +102,10 @@ size_t uxr_read_udp_data_platform(
     tv.tv_sec = timeout / 1000;
     tv.tv_usec = (timeout % 1000) * 1000;
 
-    setsockopt(platform->fd, SOL_SOCKET, SO_RCVTIMEO, &tv, sizeof(tv));
+    if (0 != setsockopt(platform->fd, SOL_SOCKET, SO_RCVTIMEO, &tv, sizeof(tv)))
+    {
+        return 0;
```
Comment (k7g03z) [inline src/c/profile/transport/ip/udp/udp_transport_posix_nopoll.c:107]: What about setting `*errcode = 1;` like below?

File: src/c/profile/transport/ip/tcp/tcp_transport_posix_nopoll.c [hunk 2]
```diff
@@ -102,7 +102,10 @@ size_t uxr_read_tcp_data_platform(
     tv.tv_sec = timeout / 1000;
     tv.tv_usec = (timeout % 1000) * 1000;
 
-    setsockopt(platform->fd, SOL_SOCKET, SO_RCVTIMEO, &tv, sizeof(tv));
+    if (0 != setsockopt(platform->fd, SOL_SOCKET, SO_RCVTIMEO, &tv, sizeof(tv)))
+    {
+        return 0;
```
Comment (k7g03z) [inline src/c/profile/transport/ip/tcp/tcp_transport_posix_nopoll.c:107]: What about setting `*errcode = 1;` like below?

**Review by k7g03z — COMMENTED** [review]
initial review of changes. Did not test their effect in the system.

File: src/c/profile/transport/ip/udp/udp_transport_posix_nopoll.c (hunk 1 above)
Comment (pablogs9) [reply src/c/profile/transport/ip/udp/udp_transport_posix_nopoll.c:108]: 75b8d1d3f222d2e1158e0e119d7ee3ffd5442805

File: src/c/profile/transport/ip/tcp/tcp_transport_posix_nopoll.c (hunk 2 above)
Comment (pablogs9) [reply src/c/profile/transport/ip/tcp/tcp_transport_posix_nopoll.c:108]: 75b8d1d3f222d2e1158e0e119d7ee3ffd5442805

**Comment by pablogs9** [comment]
Please recheck

**Comment by richiprosima** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/Micro-XRCE-DDS-Client%20Manual%20Linux/565/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/Micro-XRCE-DDS-Client%20Manual%20Windows/583/)

**Comment by k7g03z** [comment]
LGTM, I followed the code and `errcode` is checked up the call stack, so definitely necessary.

**Review by Acuadros95 — APPROVED** [review]

**Pull request merged**